package.json

```diff
@@ -2,10 +2,12 @@
     "name": "rustPlusPlus",
     "version": "1.1.0",
     "description": "A NodeJS Discord Bot that uses the rustplus.js library to utilize the power of the Rust+ Companion App with additional Quality-of-Life features.",
-    "main": "index.js",
+    "main": "index.ts",
     "scripts": {
-        "run": "node .",
-        "preinstall": "npx npm-force-resolutions"
+        "run": "ts-node .",
+        "preinstall": "npx npm-force-resolutions",
+        "test": "tsc --noEmit -p ."
```

Comment:
This will just validate your typescript is correct, also added a workflow to test it on PRs and pushes.